tcpclient/connectionsManagement.go

```diff
@@ -44,7 +45,7 @@ func TCPConnect(id int, host string, port int, wg *sync.WaitGroup, debugOut io.W
 
 	defer conn.Close()
 	connectionDescription.status = connectionEstablished
-	connectionDescription.metrics.timeTCPEstablished = time.Now()
+	connectionDescription.metrics.processingTime = time.Now().Sub(timeTCPInitiatied)
```

Comment:
I think we can also record the "processing time" as time from dialing to getting a connection Error. As a result, Errors will not impact our metrics with "0" values (so making the processing times to look nicer than they actually are). So then, if we record in more than one line the processingTime, I'd recommend to generate a simple function called "trackProcessingTime", where you do the calculation and store

Reply:
I'm not sure to mix both processing times (Established and Error)... because this metric gives idea of the TCP latency and an error could come from a reset connection from the other end or from a drop. In case of the first case (RST) we will get the same latency measurement and it could make sense to compare with the Established time, but on the other side, if it's a drop, the timeout will affect a lot on the timing. 
So I'd say that makes more sense to rename it to `establishingTime` and only take into account the established connections to calculate the metrics (average, etc.)

Reply:
What we do agree is a report with a failed connection should not consider its processing time as 0, as it previously happened, right? It screws valid-connection figures (looks as the server is replying quicker if you have an error but the rest as successful connections)

Reply:
In regards of effects of "timeout" errors... I don't see a problem to also report errored connections time, also assuming timeouts will make response times to increase a lot (better than an error makes perception worse than better). But, if it really hurts your eyes:
* IF there's successful responses, we print the processing time stats filtered by successful responses (adapted text so the reader understands the metric is only for successes)
* if there's errors,  we also print the processing time stats filtered by unsuccessful  responses (adapted text so the reader understands the metric is only for errors)
So a test with a mix of successful/unsuccessful response, will have two ending reports. One with error, only one, for errors. One with successes, only one, for successes. In the future, maybe we can make the error-by-timeout vs error-by-rst differentiation.

Reply:
I've done this:
```
type connectionMetrics struct {
	tcpEstablishedDuration time.Duration
	tcpErroredDuration     time.Duration
}
```
So, we are able to handle each case (or both) when printing the final report. I think is good to differentiate them, if not we get this 5 seconds per connection that masks the results